staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go

```diff
@@ -499,9 +514,38 @@ func (g *Cloud) GetLabelsForVolume(ctx context.Context, pv *v1.PersistentVolume)
 	}
 
 	// If the zone is already labeled, honor the hint
+	name := pv.Spec.GCEPersistentDisk.PDName
 	zone := pv.Labels[v1.LabelZoneFailureDomain]
 
-	labels, err := g.GetAutoLabelsForPD(pv.Spec.GCEPersistentDisk.PDName, zone)
+	var disk *Disk
```

Comment:
can this logic be pulled out into a named function? It seems to "get disk with optional zone" or something.

Reply:
theres another function later on that has a suspiciously similar name and interface `g.manager.GetDiskFromCloudProvider(zone, name)`

Reply:
I pulled out the logic into a separate function (getDiskByNameAndOptionalLabelZones) g.manager.GetDiskFromCloudProvider cannot be used directly as it works only for zonal Disks and only when zone is set.